### ISSUE TITLE
Little more clarity on win32 packaging?

### ISSUE THREAD

**Issue by leviking** [daid/LegacyCura#897]
Hi there, I'm trying to use the package.sh script on Windows and I'm using git bash, have the latest version of 7zip, mingw32, and nsis. I'm fairly certain I've appropriately added them to the path but I'm getting the following output: "Failed to package NSIS Installer" I've even verified that flash.sh is pointing to the appropriate directory and that it is in the path. Any thoughts? (I'm on Windows Server because I don't use Windows normally)

![failed to package](https://cloud.githubusercontent.com/assets/8175785/3594220/12445ca0-0c97-11e4-9ac1-a279ce813c00.png)

**Comment by daid** [comment]
The NSIS script outputs to the log.txt, so that might help in finding out what goes wrong.

**Comment by leviking** [comment]
Okay I looked through it and this is what I found:

Processing config: c:\Program Files (x86)\NSIS\nsisconf.nsh
Processing script file: "scripts/win32/installer.nsi" (ACP)
ReserveFile: "c:\Program Files (x86)\NSIS\Plugins\InstallOptions.dll" -> no files found.
Usage: ReserveFile [/nonfatal] [/r] [/x filespec [...]] file [file...] | [/nonfatal] /plugin file.dll
Error in script "scripts/win32/installer.nsi" on line 119 -- aborting creation process

So I reinstalled NSIS latest stable version, and it seemed to work. (I thought I tried that yesterday, but a new day always helps)

Thanks Daid!

**Comment by leviking** [comment]
Hi Daid,

After some additional testing I've found that if I just run package.sh with all of the dependencies (that I can see) squared away, the engine doesn't seem to exist after I install Cura. (It was difficult to find because previously it was using the engine from an older install).

The symptom is the save button is greyed out and there doesn't appear to be any slicing going on at all. If I run diff on the installed files the Cura/slice/ directory doesn't seem to exist at all.

Do you have any thoughts?

**Comment by leviking** [comment]
Okay, I'm fairly certain the Cura/slice directory is a red herring, at least for me.

I did find that when I tried to run the CuraEngine.exe by itself, the one I created with package.sh failed because it was missing two DLLs, libgcc_s_dw2-1.dll and libstdc++-6.dll. Both of them were in the MinGW/bin/ directory and when I copied them into the "Program Files (x86)/Cura/ directory it works without any bugs thusfar.

So I changed the CuraEngine makefile to include static linking and commented out the part of package.sh that removes and re-clones the engine repo. And, so far so good.
